src/lotto/LottoMaker.js

```diff
@@ -0,0 +1,33 @@
+import { LOTTO_MIN_PRICE, LOTTO_UNIT_PRICE } from "../constant/constants.js";
+import { inValidMessages } from "../constant/message.js";
+import { RandomNumberMaker } from "./RandomNumberMaker.js";
+import Lotto from "./Lotto.js";
+
+export class LottoMaker {
+  makeLotto(price) {
+    this.#validate(price);
+    return this.#createLotto(price);
+  }
+
+  #validate(price) {
+    this.#validateMinPrice(price);
+    this.#validateDivisible(price);
+  }
+
+  #validateMinPrice(price) {
+    if (price < LOTTO_MIN_PRICE) throw new Error(inValidMessages.minPrice);
+  }
+
+  #validateDivisible(price) {
+    if (price % LOTTO_UNIT_PRICE !== 0) throw new Error(inValidMessages.priceUnit);
+  }
+
+  #createLotto(price) {
+    let myLottoArr = [];
+    for (let i = 0; i < (price / LOTTO_UNIT_PRICE); i++) {
+      const randomNumbers = new RandomNumberMaker().makeRandomNumbers();
+      myLottoArr.push(new Lotto(randomNumbers));
+    }
+    return myLottoArr;
+  }
+}
```

Comment:
for문 대신에 고차함수를 작성하는 건 어떨까요?
for 루프를 제거하면서 코드가 더 간결하고 가독성 높일 수가 있어요!
```suggestion
#createLotto(price) {
  return Array.from({ length: price / LOTTO_UNIT_PRICE }, () => {
    const randomNumbers = new RandomNumberMaker().makeRandomNumbers();
    return new Lotto(randomNumbers);
  });
}
```

Reply:
Array.from을 생각 못했네요 꼼꼼하게 봐주셔서 감사합니다!! ^_^